addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1509,7 +1509,7 @@ var FieldEmail = InputField.extend({
      */
     _renderReadonly: function () {
         if (this.value) {
-            this.$el.text(this.value)
+            this.$el.closest("." + this.className).text(this.value)
```

Comment:
In general and by default a fix must be located a the right place (ie: here it would be in FieldPhone, and event in the SMS module) -- probably at the price of a small factorization like, for example `resetElementText()`.
This fix is quite small, so..... maybe what i'm saying is overkill.

in all logic, the same must be done in the `else` part, doesn't it.

Last but not least, even for small fixes or minor functional disturbances (though not so minor, since there is a bug report) a test is absolutely necessary. In 14.0 more than ever because we are refactoring the whole beast.

Reply:
Should be fine to keep `else` part as is, because we do want remove both elements (phone number and sms button).

I'd prefer to keep the fix here. First, to don't make refactoring in stable. Second, other field classes inherited from this one would need the fix too (imagine a SkypeField, which adds "Call" button that opens Skype app).

I'll come back with test later

Reply:
The test is ready